### ISSUE TITLE
Setting a base run before performing a model run

### ISSUE THREAD

**Issue by barnzilla** [mattwarkentin/openmpp#7]
Hi @mattwarkentin,

I'm now using `openmpp` in a Linux cloud environment and happy to report that it's working well!

One issue I'm having though: I cannot perform a model run with `openmpp` immediately after compiling a model because I do not have a base model run. Currently, `openmpp` will abort if the workset object does not have a `BaseRunDigest` parameter that is set.

Would you be able to either make the `BaseRunDigest` requirement optional or remove it altogether?

When I remove the highlighted lines below from [`ClassWorkset.R`](https://github.com/mattwarkentin/openmpp/blob/main/R/ClassWorkset.R), I'm able to successfully perform a model run without having a base run:

![image](https://github.com/user-attachments/assets/913cce48-d2b5-464f-a0d4-57c945b440a1)

Take care,
Joel

**Comment by mattwarkentin** [comment]
Thanks for opening the issue. Just so I can understand more, you don't need to reference a base model run because the compiled model includes all of the input parameters?

**Comment by barnzilla** [comment]
Yes, that's correct, Matt. I was chatting with Steve and Anatoly this afternoon and this is a feature of OpenM++ (as opposed to Modgen). A compiled model will have a base scenario (but not a base run).

**Comment by mattwarkentin** [comment]
Okay, then I think it makes sense to make this optional. Let me think a little bit about how to make this change to respect backward compatibility. But generally happy to make this change.

**Comment by barnzilla** [comment]
Much appreciated, Matt! And no rush. I have a work-around that I'm using right now.

**Comment by mattwarkentin** [comment]
Sounds good. And thanks for reporting these bugs. I only get to see how `openmpp` is used by the users here in our group, so I definitely designed the package to be opinionated for our use cases but I want it to be flexible enough for any OpenM++ user.

**Comment by mattwarkentin** [comment]
@barnzilla I am getting around to addressing this issue. Do you think it would be enough to change `rlang::abort()` to `rlang::warn()`. This warning will still be helpful for most users, but it won't halt the run for you or others who do not need a base run. You can silence this warning using normal R machinery for silencing warnings.

Basically the code would look like this:
```r
if (rlang::is_null(self$BaseRunDigest) | nchar(self$BaseRunDigest) == 0) {
  rlang::warn('Cannot find base run. Consider setting a base run with `$set_base_digest()`.')
} else {
  opts$Opts$OpenM.BaseRunDigest = self$BaseRunDigest
}
```

What do you think?

**Comment by barnzilla** [comment]
Hey @mattwarkentin, that works for me! I think that's a great solution that should keep everyone's code from breaking. Thanks again!
